pkg/roachprod/vm/aws/aws.go

```diff
@@ -1737,3 +1772,138 @@ func (p *Provider) ListLoadBalancers(*logger.Logger, vm.List) ([]vm.ServiceAddre
 	// This Provider has no concept of load balancers yet, return an empty list.
 	return nil, nil
 }
+
+func (p *Provider) GetVMArchitecture(l *logger.Logger, v *vm.VM) (vm.CPUArch, error) {
+
+	// If VM is tagged with arch FIPS, we use it
+	archLabel, ok := v.Labels["arch"]
+	if ok && archLabel == "fips" {
+		return vm.ArchFIPS, nil
+	}
+
+	// Otherwise, we query the CPU architecture from the machine type
+	az, ok := p.Config.AZByName[v.Zone]
+	if !ok {
+		return vm.ArchUnknown, fmt.Errorf("no region in %v corresponds to availability zone %v",
+			p.Config.regionNames(), v.Zone)
+	}
+
+	return p.getCPUArchFromMachineType(l, v.MachineType, az.Region.Name, WideCacheKey)
```

Comment:
Since we have the region name, shouldn't be `NarrowCacheKey`?

Reply:
Since we're describing an existing VM, we' re not trying to check if this machine type exists in this region (it obviously exists, since the VM runs), but only want to get the architecture from the machine type, and using WideCacheKey allows to use the result from the same machine-type previously grabbed from another region.

It speeds things up by a lot (15s in some tests).